samtranslator/model/api/http_api_generator.py

```diff
@@ -344,12 +344,9 @@ def _construct_basepath_mappings(self, basepaths, http_api):
                     path = ""
                 else:
                     # ignore leading and trailing `/` in the path name
-                    m = re.search(r"[a-zA-Z0-9]+[\-\_]?[a-zA-Z0-9]+", path)
-                    path = m.string[m.start(0) : m.end(0)]
-                    if path is None:
-                        raise InvalidResourceException(self.logical_id, "Invalid Basepath name provided.")
+                    path = path.strip("/")
```

Comment:
I think you can actually replace the whole if/else.
if `path == "/"`, `path.strip("/")` will result in `""`.
So you can replace the whole L343-L347 with `path = path.strip("/")`